library/x509_crt.c

```diff
@@ -2951,23 +2951,138 @@ static int x509_crt_verify_chain(
     }
 }
 
+static int parse_ipv4( const char* h, char *addr )
+{
+    unsigned char ip[4];
+    int i;
+    const char *strt = h;
+    char *endp;
+    unsigned long v;
+    for( i = 0; i < 4; i++ )
+    {
+        v = strtoul( strt, &endp, 10 );
+        if( endp == strt || v > 255 )
+        {
+            return ( 0 );
+        }
+
+        if( i < 3 && *endp != '.' )
+        {
+            return ( 0 );
+        }
+        ip[i] = v & (unsigned char)0xFF;
+        strt = endp + 1;
+    }
+
+    memcpy( addr, ip, sizeof ip);
+    return ( 1 );
+}
+
+static int parse_ipv6( const char* h, size_t hlen, char *addr )
```

Comment:
Could you add a comment explaining what the return value of `parse_ipv6()` means? Hopefully it matches the same return style as `x509_crt_check_cn()`.